test/integration/targets/azure_rm_sqlserver/tasks/main.yml

```diff
@@ -0,0 +1,105 @@
+- name: Prepare random number
+  set_fact:
+    random_postfix: "{{ 1000 | random }}"
+  run_once: yes
+
+- name: Delete unexisting instance of SQL Server -- check mode
+  azure_rm_sqlserver:
+    resource_group: "{{ resource_group }}"
+    name: "sqlsrv{{ random_postfix }}{{ resource_group | hash('md5') | truncate(7, True, '') }}"
+    state: absent
+  check_only: yes
+  register: output
+- name: Assert the state has changed
+  assert:
+    that:
+      - output.changed == false
+
+- name: Delete unexisting instance of SQL Server
```

Comment:
Don't need this test as you should be starting with a random resource group so no SQL server would exist anyway